### PR TITLE
Lazy vertex/index buffer application in GL

### PR DESCRIPTION
**Pull request by Kangz**
PTAL

### CODE REVIEW

File: src/backend/opengl/InputStateGL.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        glBindVertexArray(vertexArrayObject);
+        auto& attributesSetMask = GetAttributesSetMask();
+        for (uint32_t location = 0; location < attributesSetMask.size(); ++location) {
+            if (!attributesSetMask[location]) {
+                continue;
+            }
+            auto attribute = GetAttribute(location);
+            glEnableVertexAttribArray(location);
+
+            attributesUsingInput[attribute.bindingSlot][location] = true;
```
Comment (Kangz) [inline src/backend/opengl/InputStateGL.cpp]: This line is the only change in this file.

File: src/backend/opengl/CommandBufferGL.cpp [hunk 2]
```diff
@@ -85,9 +109,87 @@ namespace opengl {
...
+        };
+
+        // Vertex buffers and index buffers are implemented as part of an OpenGL VAO that corresponds to an
+        // InputState. On the contrary in NXT they are part of the global state. This means that we have to
+        // re-apply these buffers on an InputState change.
+        struct InputBufferTracker {
+            bool indexBufferDirty = false;
+            Buffer* indexBuffer = nullptr;
+
+            std::bitset<kMaxVertexAttributes> dirtyVertexBuffers;
```
Comment (kainino0x) [inline src/backend/opengl/CommandBufferGL.cpp]: Why `kMaxVertexAttributes` here and `kMaxVertexInputs` below? Seems like they're both indexed the same way in `OnSetVertexBuffers`.

Comment (Kangz) [reply src/backend/opengl/CommandBufferGL.cpp]: Whoops, code was wrong. Done.

**Comment by Kangz** [comment]
Thanks for the review!

**Pull request merged**